### ISSUE TITLE
TimesNet's wrong code.

### ISSUE THREAD

**Issue by QiSu77** [thuml/Time-Series-Library#466]
"def FFT_for_Period(x, k=2):
    # [B, T, C]
    xf = torch.fft.rfft(x, dim=1)
    # find period by amplitudes
    frequency_list = abs(xf).mean(0).mean(-1)
    frequency_list[0] = 0
    _, top_list = torch.topk(frequency_list, k)
    top_list = top_list.detach().cpu().numpy()
    period = x.shape[1] // top_list
    return period, abs(xf).mean(-1)[:, top_list]"
I wonder that the topk function returns  value and index of the top values in the tensor.   However, the code uses the index to get the period, is it wrong?

**Comment by wuhaixu2016** [comment]
Hi, it is correct. For example, the index i refers to the frequency value is i, you can obtain the period by period/frequency.

It is clearer, if you read the official definition of torch.fft: https://pytorch.org/docs/stable/generated/torch.fft.fft.html#torch.fft.fft
